### PR TITLE
HW #6 (Monolith and microservices) 

### PR DESCRIPTION
**Pull request by eluyema**
Devide moniloth to 3 microservices and api gateway:
- Exchange rate
- Mailer
- Subscription
- ApiGateway

microservices drawio

### CODE REVIEW

**Comment by eluyema** [comment]
For review it's better to pull branch and look at structure of microserves in IDE

**Comment by cholombytko** [comment]
I guess it would be better to use switch case or mapping here
https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-eluyema/blob/e7953e9cdffab5650bc32fc0b3d9929ab7d42c65/api-gateway/src/infrastructure/http/subscription.service.ts#L42-L56

**Comment by cholombytko** [comment]
In http of infrastructure layer of exchange-rate service, i would create separate folder for decorators, but it's only cosmetic change

**Comment by cholombytko** [comment]
Any other changes looks good to me, great job

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ For mail template it uses handlebars templates.
 
 How to start application?
 
-1. You need to create .env file (look at example .env.sample)
+1. You need to create .env file in each microservice (look at example .env.sample)
```
Comment (Larisa-Antsifrova) [inline README.md:7]: Once again: that's a valuable skill to keep documentation up to date :)

File: subscription/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Stay in touch
+
+- Author - [Kamil Myśliwiec](https://kamilmysliwiec.com)
+- Website - [https://nestjs.com](https://nestjs.com/)
+- Twitter - [@nestframework](https://twitter.com/nestframework)
+
+## License
+
+Nest is [MIT licensed](LICENSE).
```
Comment (Larisa-Antsifrova) [inline subscription/README.md:73]: Probably this boilerplate is not needed in the project :)

**Pull request merged**